packages/store/src/base.ts

```diff
@@ -114,7 +114,9 @@ export class BaseBlockModel<Props = unknown>
   text?: TextType;
   sourceId?: string;
 
+  // TODO: separate from model
   parentIndex?: number;
+  index?: number;
```

Comment:
Can we try to to hold this field (seems to be used for UI query caching for now) outside of `BaseBlockModel`? For example, is it possible to associate it with another cache (mapping) in `SelectionManager`?

Since the `index` is not set into CRDT side, my concern is that after some undo/redo operations, there may be stale `index` state that could lead to hard-to-track issues.